payment/src/main/java/com/yas/payment/service/PaymentProviderService.java

```diff
@@ -1,31 +1,113 @@
 package com.yas.payment.service;
 
 import com.yas.commonlibrary.exception.NotFoundException;
+import com.yas.payment.mapper.CreatePaymentProviderMapper;
+import com.yas.payment.mapper.PaymentProviderMapper;
+import com.yas.payment.mapper.UpdatePaymentProviderMapper;
 import com.yas.payment.model.PaymentProvider;
 import com.yas.payment.repository.PaymentProviderRepository;
 import com.yas.payment.utils.Constants;
-import com.yas.payment.viewmodel.PaymentProviderVm;
+import com.yas.payment.viewmodel.paymentprovider.CreatePaymentVm;
+import com.yas.payment.viewmodel.paymentprovider.MediaVm;
+import com.yas.payment.viewmodel.paymentprovider.PaymentProviderVm;
+import com.yas.payment.viewmodel.paymentprovider.UpdatePaymentVm;
+import java.util.Collections;
 import java.util.List;
-import lombok.RequiredArgsConstructor;
+import java.util.Map;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+/**
+ * Payment provider service.
+ */
 @Slf4j
 @Service
-@RequiredArgsConstructor
 public class PaymentProviderService {
+
+    private final MediaService mediaService;
     private final PaymentProviderRepository paymentProviderRepository;
 
+    private final PaymentProviderMapper paymentProviderMapper;
+    private final CreatePaymentProviderMapper createPaymentProviderMapper;
+    private final UpdatePaymentProviderMapper updatePaymentProviderMapper;
+
+    public PaymentProviderService(
+        MediaService mediaService,
+        PaymentProviderMapper paymentProviderMapper,
+        CreatePaymentProviderMapper createPaymentProviderMapper,
+        UpdatePaymentProviderMapper updatePaymentProviderMapper,
+        PaymentProviderRepository paymentProviderRepository
+    ) {
+        this.mediaService = mediaService;
+        this.paymentProviderMapper = paymentProviderMapper;
+        this.createPaymentProviderMapper = createPaymentProviderMapper;
+        this.updatePaymentProviderMapper = updatePaymentProviderMapper;
+        this.paymentProviderRepository = paymentProviderRepository;
+    }
```

Comment:
It looks like you replaced the `@RequiredArgsConstructor` annotation with explicit constructor code in the `PaymentProviderService` class. I was wondering if there’s a specific reason for this change, as it makes the code a bit longer. Just wanted to understand the reasoning behind it.